api/api-feeds-api-v1/src/main/java/com/thoughtworks/go/apiv1/feedsapi/FeedsApiControllerV1.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2020 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.thoughtworks.go.apiv1.feedsapi;
+
+import com.thoughtworks.go.api.ApiController;
+import com.thoughtworks.go.api.ApiVersion;
+import com.thoughtworks.go.api.spring.ApiAuthenticationHelper;
+import com.thoughtworks.go.config.exceptions.BadRequestException;
+import com.thoughtworks.go.server.service.FeedService;
+import com.thoughtworks.go.spark.RequestContext;
+import com.thoughtworks.go.spark.Routes;
+import com.thoughtworks.go.spark.spring.SparkSpringController;
+import org.dom4j.Document;
+import org.dom4j.io.OutputFormat;
+import org.dom4j.io.XMLWriter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import spark.Request;
+import spark.Response;
+
+import java.io.IOException;
+import java.io.StringWriter;
+
+import static java.lang.String.format;
+import static org.apache.commons.io.FilenameUtils.removeExtension;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.springframework.http.MediaType.APPLICATION_XML_VALUE;
+import static spark.Spark.*;
+
+@Component
+public class FeedsApiControllerV1 extends ApiController implements SparkSpringController {
+    private final ApiAuthenticationHelper apiAuthenticationHelper;
+    private final FeedService feedService;
+
+    @Autowired
+    public FeedsApiControllerV1(ApiAuthenticationHelper apiAuthenticationHelper,
+                                FeedService feedService) {
+        super(ApiVersion.v1);
+        this.mimeType = APPLICATION_XML_VALUE;
+        this.apiAuthenticationHelper = apiAuthenticationHelper;
+        this.feedService = feedService;
+    }
+
+    @Override
+    public String controllerBasePath() {
+        return Routes.FeedsAPI.BASE;
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("/*", mimeType, this::setContentType);
+            before("/*", mimeType, this.apiAuthenticationHelper::checkUserAnd403);
```

Comment:
`NormalUserSecurity`? Should this be based on some permissions (pipeline, pipeline group level view permissions)?

Not sure if keeping it public for all has any security implications.
\cc: @arvindsv @bdpiparva

Reply:
@GaneshSPatil - That is verified at the service level so I skipped that check from the controller.

Reply:
I think we have a check at the controller level to early return based on the user's permission. This makes authorization logic scattered between controller and services. Ideally, I prefer to have it at the service level. Controllers are something that can change based on libraries like sparks or languages but in most of the applications, services will remain in the same(changing services mean rewriting entire application).